src/movegen.h

```diff
@@ -21,6 +21,8 @@
 #ifndef MOVEGEN_H_INCLUDED
 #define MOVEGEN_H_INCLUDED
 
+#include <algorithm>
+
```

Comment:
@snicolet thanks for the review. 

I'm following the style used throughout the code where there is a blank line between C++ system headers and project headers. That's a common style as well (e.g. https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes )